lib/engine/step/half_pay.rb

```diff
@@ -4,9 +4,13 @@ module Engine
   module Step
     module HalfPay
       def half(entity, revenue)
-        withheld = (revenue / 2 / entity.total_shares).to_i * entity.total_shares
+        withheld = half_pay_withhold_amount(entity, revenue)
         { corporation: withheld, per_share: payout_per_share(entity, revenue - withheld) }
       end
+
+      def half_pay_withhold_amount(entity, revenue)
```

Comment:
I'm not sure if adding a non-publicly-exposed method in a module in Ruby is kosher.